amazonka/src/Control/Monad/Trans/AWS.hs

```diff
@@ -249,8 +249,7 @@ instance MonadBaseControl b m => MonadBaseControl b (AWST' r m) where
     restoreM     = defaultRestoreM
 
 instance MonadUnliftIO m => MonadUnliftIO (AWST' r m) where
-    askUnliftIO = AWST' $ (\(UnliftIO f) -> UnliftIO $ f . unAWST)
-        <$> askUnliftIO
+    withRunInIO inner = (AWST' $ (\(UnliftIO f) -> UnliftIO $ f . unAWST) <$> askUnliftIO) >>= \u -> liftIO (inner (unliftIO u))
```

Comment:
Basically inlined the old definition of `withRunInIO`: https://hackage.haskell.org/package/unliftio-core-0.1.2.0/docs/src/Control.Monad.IO.Unlift.html#withRunInIO

There may be a better way to write this.